2024/opmodes/TwoStickTeleop.java

```diff
@@ -185,38 +194,25 @@ public void handleEvent(RobotEvent e) {
         });
 
         //Gamepad 2
-        this.addTask(liftMotorTask);
+        //this.addTask(liftMotorTask);
 
         this.addTask(new GamepadTask(this, GamepadTask.GamepadNumber.GAMEPAD_2) {
             //@Override
             public void handleEvent(RobotEvent e) {
                 GamepadEvent gamepadEvent = (GamepadEvent) e;
 
                 switch (gamepadEvent.kind) {
-
-                    case BUTTON_X_DOWN:
-                        //position 0
-                        armServo.setPosition(ARM_FRONT);
-                        break;
-                    case BUTTON_B_DOWN:
-                        //position 1
-                        armServo.setPosition(ARM_BACK);
-                        break;
-                    case BUTTON_A_DOWN:
-                        //position 1
-                        coneServo.setPosition(CONE_GRAB);
-                        break;
                     case BUTTON_Y_DOWN:
-                        //position 0 (original pos)
-                        coneServo.setPosition(CONE_RELEASE);
-                        break;
-                    case LEFT_TRIGGER_DOWN:
-                        //position 1
-                        junctionAligner.setPosition(ALIGNER_FRONT);
+                        // set arm to extend to its highest capacity to lift robot
+                        hangingMotor.setTargetPosition(HANGING_FULLY_EXTENDED);
+                        // encoder allows you to know how much the motor has spun (distance)
+                        hangingMotor.setMode(DcMotor.RunMode.RUN_TO_POSITION);
```

Comment:
This is fine, but RunToEncoderValueTask is designed to accomplish just this functionality while also telling you where it is and when it is finished through the event callbacks.

Reply:
Based on a quick read of RunToEncoderValueTask.java, it looks like each instantiation of the task goes through the `STOP_AND_RESET_ENCODER`, `setTargetPosition`, `RUN_TO_POSITION` cycle, resetting the absolute position each time, which I think is not desirable.